### ISSUE TITLE
Language problems in 4.1.0

### ISSUE THREAD

**Issue by christaikobo** [arsenetar/dupeguru#760]
First of all, thank you for this great project and all the awesome work!

I just installed 4.1.0 on windows, the language only defaults to system language,

e.g., if my system is in English, the interface is displayed in English, changing it in display setting seems to be doing nothing, even if I reboot my PC, the displayed language is still English. Same applies when I installed it in a Chinese system, language defaults to Chinese, there is simply no way to change it. 

Another issue is that on [transifex](https://www.transifex.com/hsoft/dupeguru/), I saw that Chinese translation is 98% finished with only 4 strings to be translated, however in reality more than half of the interface is not translated. 

 - OS: Windows 10 1903 x64
 - Version: 4.1.0

![1](https://user-images.githubusercontent.com/24227628/104262899-7ef0d380-54c3-11eb-9081-3ebeec3ceabb.png)

**Comment by arsenetar** [comment]
@terrytw The transifex was from the previous maintainer, it has not been updated with the latest files in quite some time.  A good portion of the items in the preference window are new so that would explain the lack of translations.  

I will look into why the language is not changing as I can reproduce.

**Comment by arsenetar** [comment]
This ended up being broke from the change in b44d1652b6b4c1786b56ae8d93c55a9bc6d0e06e, turns out `run.py` also creates a QSettings object its own way.  Confirmed making same changes in `run.py` resolves the issue.  Will be working on cleaning this up.

**Comment by christaikobo** [comment]
Thanks again for the swift response!

I would like to contribute by helping translating the new version, will you update the transifex page or switch to other services like weblate?

**Comment by arsenetar** [comment]
I am probably going to rebuild the 4.1.0 windows builds with this fix tomorrow some time.  I also started setting up a project here www.transifex.com/na-424/dupeguru-1 for now, I need to get all the files uploaded for what the languages have so far.  You are welcome to help with the translation.

**Comment by glubsy** [comment]
I did some translations earlier this week on another project, I'll have a look too.

**Comment by christaikobo** [comment]
The transifex link gave me a 404 error, I assume you have not finished uploaded the files yet. Feel free to @ me once you are done uploading it, so I can start working on the Chinese translation.

**Comment by arsenetar** [comment]
Ah, I ended up updating the organization, I guess that changes the URL, it is at https://www.transifex.com/voltaicideas/dupeguru-1 now, although I have reached out to hsoft about potentially transferring the old project over instead so perhaps hold off for a couple days here before looking at it, I'll update both of you once it is ready.

**Comment by arsenetar** [comment]
@terrytw & @glubsy, ends up we are going to be using https://www.transifex.com/voltaicideas/dupeguru-1 for the project.  So everything there should be setup and good to use.

**Comment by christaikobo** [comment]
I'm not too familiar with the Transifex work flow, but I think I lodged a request to join the project.

**Comment by arsenetar** [comment]
Hmm, looks like you requested adding the language Chinese (zh) to the project, the project has (zh_CN) which is ~75%, did you intend to add a new language?  I think if you want to help translate you can click the Join project button and pick the language. I see I can invite, I don't see anything aside from the language request.

**Comment by arsenetar** [comment]
Oh now I see it, not sure why it does not show me these in a more obvious location.

**Comment by christaikobo** [comment]
I think I have been added to the project. Thanks. 

I didn't mean to add a new language, was just a little confused by the interface. You can ignore the add language request. Sorry about that. 

I think I will start working on the translations in the weekend.

**Comment by glubsy** [comment]
Unfortunately Transifex is not an ideal solution. Can't translate more than 3 languages somehow. 
Also, Machine translations are often more than enough for our use case (or at least a decent enough base), and Transifex seems to be way too strict about that. 
I wonder if there is a better way of handling translation files...

Perhaps I could edit the .po files manually?

**Comment by arsenetar** [comment]
I guess I am not seeing the issue you are pointing out, are you saying you cannot be a translator on more than 3 languages, I'm pretty sure you can.  I can add you to other languages from my side for sure.

"Also, Machine translations are often more than enough for our use case (or at least a decent enough base), and Transifex seems to be way too strict about that."  - Also not seeing what you mean here, I applied some machine translated items and they seem just fine.

**Comment by glubsy** [comment]
Alright, nevermind. From my side, I could not translate languages which are not tied to my profile somehow. But I see now I can request to join a team for each language. I guess I'll do that.

By machine translation I meant the same thing as above, having to request to join a team in order to supply translations, even if not a native speaker. Not very practical for projects where contributors are very scarce. Machine translation is often good enough, so let's just allow anyone to submit flaky machine translations. If native speakers want to improve the files, they are welcome to do so.

**Comment by glubsy** [comment]
Actually it's still weird, I can't join some languages which are missing translators (Portuguese Brazil for example).

**Comment by arsenetar** [comment]
@glubsy , I think I just have to approve the join, you should be good for the five requests I saw.

**Comment by christaikobo** [comment]
@arsenetar Just finished the translation. A few questions:

1. In `ui.pot`, No. 112 and No. 118 both have `Results` as text, I assumed in 112, `{}` will be placed by the number of results, whereas in 118, `%@` will be replaced by the description of the result. Please correct me if I assumed wrong. 

2. I think you made a typo in the original text of No. 184 from `ui.pot`, I think you meant to say 

> 2. Regular expressions with path separator in them will be compared to the full path to the file

instead of 

> 2. Regular expressions with no path separator in them will be compared to the full path to the file

**Comment by glubsy** [comment]
> 2\. I think you made a typo in the original text of No. 184 from `ui.pot`, I think you meant to say

This is fixed in #775 .
